client/src/app/party-select/party-select.component.html

```diff
@@ -1,24 +1,58 @@
 <div class="party-select-container">
-	<nz-avatar class="self-icon" [nzSize]="64"
-	    nzSrc="{{jobs.get(selfJob)!.path}}" (click)="changeSelf()">
-	</nz-avatar>
-	<nz-avatar class="party-member-icon" *ngFor="let member of party; index as i;"
-	    [nzSize]="64" nzSrc="{{jobs.get(member)!.path}}" (click)="changeParty(i)">
-	</nz-avatar>
+  <nz-avatar class="self-icon" [nzSize]="64"
+      nzSrc="{{jobConsts.JOBS.get(selfJob)!.path}}" (click)="changeSelf()">
+  </nz-avatar>
+  <nz-avatar class="party-member-icon" *ngFor="let member of party; index as i;"
+      [nzSize]="64" nzSrc="{{jobConsts.JOBS.get(member)!.path}}"
+      (click)="changeParty(i)">
+  </nz-avatar>
 </div>
 <div class="buffs-container">
-	<img class="buff-icon" *ngFor="let buff of jobs.get(selfJob)!.buffs;"
+  <img class="buff-icon" *ngFor="let buff of jobConsts.JOBS.get(selfJob)!.buffs;"
+      src="{{buffs.get(buff)!.path}}"/>
+  <ng-container *ngFor="let member of party;">
+    <img class="buff-icon"
+	    *ngFor="let buff of jobConsts.JOBS.get(member)!.buffs;"
 		src="{{buffs.get(buff)!.path}}"/>
-	<ng-container *ngFor="let member of party;">
-		<img class="buff-icon" *ngFor="let buff of jobs.get(member)!.buffs;"
-		    src="{{buffs.get(buff)!.path}}"/>
-	</ng-container>
+  </ng-container>
 </div>
 <nz-modal #jobPickerModal [(nzVisible)]="jobPickerVisible" [nzFooter]="null"
-          (nzOnCancel)="closeJobPicker()">
+    (nzOnCancel)="closeJobPicker()">
   <ng-container *nzModalContent>
-    <nz-avatar class="party-member-icon" *ngFor="let job of jobs.values();"
-	    [nzSize]="64" nzSrc="{{job.path}}" (click)="pickJob(job.name)">
-	</nz-avatar>
+  	<div class="role-container">
+  	  <div class="role-name">Tank</div>
+  	  <nz-avatar class="party-member-icon"
+          *ngFor="let job of jobConsts.TANK_JOBS;"
+	      [nzSize]="64" nzSrc="{{job.path}}" (click)="pickJob(job.name)">
+	  </nz-avatar>
+  	</div>
+  	<div class="role-container">
+  	  <div class="role-name">Healer</div>
+  	  <nz-avatar class="party-member-icon"
+          *ngFor="let job of jobConsts.HEALER_JOBS;"
+	      [nzSize]="64" nzSrc="{{job.path}}" (click)="pickJob(job.name)">
+	  </nz-avatar>
+  	</div>
+  	<div class="role-container">
```

Comment:
i think for the dps jobs there's no need to enforce roles. people run all sorts of interesting comps nowadays (and the dps sim will handle calculating the buff % for having different roles)

Reply:
Yeah, I don't have any role enforcement going on right now (you can make a party of 8 PLD if that's what you really want), this is more to make it easier to find what job you're looking for.

Reply:
ack, i misunderstood what this code was supposed to do